packages/eas-cli/src/build/build.ts

```diff
@@ -161,38 +171,33 @@ export async function prepareBuildRequestForPlatformAsync<
   };
 }
 
-const SERVER_SIDE_DEFINED_ERRORS = [
-  'TURTLE_DEPRECATED_JOB_FORMAT',
-  'EAS_BUILD_FREE_TIER_DISABLED',
-  'EAS_BUILD_FREE_TIER_DISABLED_IOS',
-  'EAS_BUILD_FREE_TIER_DISABLED_ANDROID',
-  'VALIDATION_ERROR',
-];
+const SERVER_SIDE_DEFINED_ERRORS: { [key: string]: any } = {
```

Comment:
```suggestion
const SERVER_SIDE_DEFINED_ERRORS: { [key: string]: Error } = {
```

Reply:
TS does complain about that - it expects Error instance, not Error type

Reply:
`:typeof Error` then?

Reply:
`TS2322: Type 'typeof TurtleDeprecatedJobFormatError' is not assignable to type 'ErrorConstructor'.   Type 'typeof TurtleDeprecatedJobFormatError' provides no match for the signature '(message?: string | undefined): Error'. `

Reply:
I think `Record<string, typeof Error>` should work fine

Reply:
I checked out your branch locally and `new () => Error` is what should work here. You'll have to change the type for `ErrorClass` a few line below.

Reply:
@dsokal I've seen in our codebase that we usually prefer to type such objects as records over explicit declarations. Is this some sort of rule we want to stick to or is it just coincidence and we are fine with both ways of typing such a thing?

Reply:
You're right, `Record<...>` is always better.

Reply:
`Record<string, typeof Error>` didn't work, but `Record<string, typeof EasCommandError>` did 🤷 
All these error types are sub-types of `EasCommandError` so it's even better/more specific